src/utils/HiddenRoute.js

```diff
@@ -0,0 +1,22 @@
+import { Component } from 'react';
+import { Redirect, Route } from 'react-router-dom';
+
+/*
+    This Component simply Hides a component if the user is logged in. IE hiding the longin link when user is logged in already.
+  */
+const HiddenRoute = ({ component: Component, ...rest }) => {
+  return (
+    <Route
+      {...rest}
+      render={() => {
+        if (!localStorage.getItem('token')) {
+          return <Component />;
+        } else {
+          return <Redirect to="/" />;
```

Comment:
I like that it redirects to the root '/' path so that the user is redirected to their own user dashboard. Its a better way to do it than checking for the role again and redirecting based on that result.